<?xml version="1.0" ?>
<launch>



    <!-- TODO: compensate z-position between base and dvl -->
    <!-- day_10 test_2: init rov_depth 1.7 + 0.391 -->
    <!-- day_11 test_2: init rov_depth 2.3 + 0.391 -->
    <!-- <arg name="rov_depth" default="4.471" /> -->

    <!-- Convert into global frame -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="tf_earth" 
          args="0 0 -$(arg rov_depth) 0 0 0 1 odom_earth odom 100" /> -->

    <!-- Pre-processing DVL: correct sound speed, filter noise, depth measurement, depth initialization? -->
    <param name="air_pressure" value="0.377999"/>
    <param name="standard_soundSpeed" value="1480"/>
    <param name="z_B_D" value="0.391"/>
    <param name="filter_velocity" value="false"/>
    <node name="proc_dvl_node" pkg="rov_processing" type="proc_dvl_serial"  output="screen">
        <remap from="/buttom_track" to="/rov/sensors/dvl/bottom_track"/>
        <remap from="/current_profile" to="/rov/sensors/dvl/current_profile"/>
    </node>


    <!-- Pre-processing IMU: filter raw imu using madgwick filter -->
    <node name="madgwick_imu" pkg="imu_filter_madgwick" type="imu_filter_node" output="screen">
        <remap from="/imu/data_raw" to="/rov/sensors/ahrs/imu/data"/>
        <remap from="/imu/mag" to="/rov/sensors/ahrs/mag" />
        <remap from="/imu/data" to="/rov/processed/ahrs/imu_madgwick"/>

        <param name="use_mag" value="true"/>
        <param name="use_magnetic_field_msg" value="true"/>
        <param name="mag_bias_x" value="-0.03661794"/>  
        <param name="mag_bias_y" value="0.01874022"/>  
        <param name="mag_bias_z" value="0.00129924"/>  
        <param name="orientation_stddev" value=".005"/> 
        <param name="gain" value="0.5" />
        <param name="zeta" value="0.001" />
        <param name="publish_tf" value="false"/>
        <param name="remove_gravity_vector" value="false"/>
    </node>


    <!-- Robot Localization EKF Odometry -->
    <node name="ekf_odometry" pkg="robot_localization" type="ekf_localization_node" clear_params="true" output="screen">
        <rosparam command="load" file="$(find rov_processing)/config/robot_localization/imu_vel_dep.yaml" />
        <remap from="odometry/filtered" to="/rov/processed/robot_localization_odometry"/>
    </node>


    <!-- Utility: save DVL ice map -->
    <!-- <param name="odom_hz" value="50"/>
    <param name="map_frame" value="odom_earth"/> 
    <param name="init_depth" value="$(arg rov_depth)"/>
    <param name="save_path" value="/home/lin/develop/ros/soslab_ws/src/rov/rov_processing/result/"/>
    <node name="test_save" pkg="rov_processing" type="test_save" output="screen" >
        <remap from="odometry" to="/rov/processed/ekf_odometry"/>
        <remap from="cloud" to="/rov/processed/dvl/ice_cloud"/>
    </node>   -->

</launch>